src/include/verbose.h

```diff
@@ -6,84 +6,59 @@
-    gen(WT_VERB_API)                   \
-    gen(WT_VERB_BACKUP)                \
-    gen(WT_VERB_BLKCACHE)              \
-    gen(WT_VERB_BLOCK)                 \
-    gen(WT_VERB_CHECKPOINT)            \
-    gen(WT_VERB_CHECKPOINT_CLEANUP)    \
-    gen(WT_VERB_CHECKPOINT_PROGRESS)   \
-    gen(WT_VERB_COMPACT)               \
-    gen(WT_VERB_COMPACT_PROGRESS)      \
-    gen(WT_VERB_DEFAULT)               \
-    gen(WT_VERB_ERROR_RETURNS)         \
-    gen(WT_VERB_EVICT)                 \
-    gen(WT_VERB_EVICTSERVER)           \
-    gen(WT_VERB_EVICT_STUCK)           \
-    gen(WT_VERB_EXTENSION)             \
-    gen(WT_VERB_FILEOPS)               \
-    gen(WT_VERB_GENERATION)            \
-    gen(WT_VERB_HANDLEOPS)             \
-    gen(WT_VERB_HS)                    \
-    gen(WT_VERB_HS_ACTIVITY)           \
-    gen(WT_VERB_LOG)                   \
-    gen(WT_VERB_LSM)                   \
-    gen(WT_VERB_LSM_MANAGER)           \
-    gen(WT_VERB_METADATA)              \
-    gen(WT_VERB_MUTEX)                 \
-    gen(WT_VERB_OUT_OF_ORDER)          \
-    gen(WT_VERB_OVERFLOW)              \
-    gen(WT_VERB_READ)                  \
-    gen(WT_VERB_RECONCILE)             \
-    gen(WT_VERB_RECOVERY)              \
-    gen(WT_VERB_RECOVERY_PROGRESS)     \
-    gen(WT_VERB_RTS)                   \
-    gen(WT_VERB_SALVAGE)               \
-    gen(WT_VERB_SALVAGE_PROGRESS)      \
-    gen(WT_VERB_SHARED_CACHE)          \
-    gen(WT_VERB_SPLIT)                 \
-    gen(WT_VERB_TEMPORARY)             \
-    gen(WT_VERB_THREAD_GROUP)          \
-    gen(WT_VERB_TIERED)                \
-    gen(WT_VERB_TIMESTAMP)             \
-    gen(WT_VERB_TRANSACTION)           \
-    gen(WT_VERB_UPGRADE_PROGRESS)      \
-    gen(WT_VERB_VERIFY)                \
-    gen(WT_VERB_VERIFY_PROGRESS)       \
-    gen(WT_VERB_VERSION)               \
-    gen(WT_VERB_WRITE)
-/* clang-format on */
-
-/* Permitted verbose event categories that can be used when defining a verbose message. */
-typedef enum {
-    WT_GEN_VERBOSE_CATEGORIES(WT_GEN_VERBOSE_ENUM)
-    /* This entry needs to be the last in order to track the number of category items. */
-    WT_VERB_NUM_CATEGORIES,
-} WT_VERBOSE_CATEGORY;
-
-/* Convert a category to its string representation. */
+// clang-format off
 #define WT_VERBOSE_CATEGORY_STR(category) \
-    (const char *[]){WT_GEN_VERBOSE_CATEGORIES(WT_GEN_VERBOSE_ENUM_STR)}[category]
-
-/*
- * Permitted verbosity levels; to be used when defining verbose messages. The levels define a range
- * of severity categories, with WT_VERBOSE_ERROR being the lowest, most critical level (used by
- * messages on critical error paths) and WT_VERBOSE_DEBUG being the highest verbosity/informational
- * level (mostly adopted for debugging).
- */
-typedef enum {
-    WT_VERBOSE_ERROR = -3,
-    WT_VERBOSE_WARNING,
-    WT_VERBOSE_NOTICE,
-    WT_VERBOSE_INFO,
-    WT_VERBOSE_DEBUG
-} WT_VERBOSE_LEVEL;
+    (const char *[]){ \
```

Comment:
Declaring the string array in the macro is interesting.  It means that on each usage of the macro, potentially a new array will be created.  The macro is currently used twice.  A quick test with clang shows that two arrays are created (though the strings are shared).  With gcc, only a single array is created.  So it's a little wasteful (on clang at least), but I would be more worried if the use of the macro or the technique proliferated in the future.

Anyway, I think it would be better to create a single static array, perhaps declare it in `support/err.c`.  If you wanted to generate the strings into `verbose.h`, you could still do so, and change the macro to one that does initialization.  (kind of like how `WT_CURSOR_STATIC_INIT` does initialization, except the values could be generated).

Reply:
Thanks for your feedback @ddanderson, I have reworked this as part of 6cd5edc067feda2f65efbf6de6d17c3cbe7063d7.
We still have the definition in `verbose.h` but we create the array in `err.c`. Keeping the definition in `verbose.h` makes it easier for the script `verbose.py` to update only one file.